linker/src/lib.rs

```diff
@@ -360,9 +360,11 @@ fn namespaced_expression(namespace: String, mut expr: Expression) -> Expression
     expr.visit_expressions_mut(
         &mut |expr| {
             if let Expression::Reference(_, refs) = expr {
-                refs.path = SymbolPath::from_parts(
-                    once(Part::Named(namespace.clone())).chain(refs.path.clone().into_parts()),
-                );
+                if !refs.path.is_std() {
```

Comment:
Bug: We used to namespace `std` namespaces as well. So, if there was a lookup from `main` into `main_sub`, using something in `std::*` in its expressions, it would turn into `main::std::*`.